### PR TITLE
Fix server side query in pdf report filter

### LINKED ISSUE #5686: Server side PDF report filters are not being properly parsed
| Wazuh | OpenSearch |
| ----- | ------- | 
| 4.4.5   | 2.6     | 

**Description**

Some filters are not interpreted properly in the server side PDF report filter inteprete:
https://github.com/wazuh/wazuh-kibana-app/blob/v4.4.5-7.17.9/server/controllers/wazuh-reporting.ts#L69-L85

When generating a PDF report, the queries made in the client side work properly, but when the server receives the filters it transforms them into a query string in the `sanitizeKibanaFilters` function.

### Example

When one clicks on the `Level 12 or above alerts` KPI it applies a `range` filter between 12 and null. Then in gets transformed in the server side to rule.level: 12-null

Screenshot from 2023-07-17 18-36-52
Screenshot from 2023-07-17 16-40-51

### PDF Report result
The visualizations have the alerts, but the query to create Alerts Summary table and other KPIs have empty results.
image

**Preconditions**
1. Have indexed alerts

**Steps to reproduce**
1. Navigate to Security Alerts
2. Click on `Level 12 or above alerts` number
3. Click on `Generate report`
4. The PDF report "Alerts Summary" should not have any results

### PR DESCRIPTION
**Pull request by asteriscos**
### Description
Hi team,
this pull request fixes the server side query in pdf report filter.
 
### Issues Resolved
#5686 

### Evidence
[Provide screenshots or videos to prove this PR solves the issues]

### Test

Legend:
:black_circle:: none
:green_circle:: pass
:yellow_circle:: warning
:red_circle:: fail
:white_circle:: not applicable

## UI

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| Generate module report in Security events module without filters set by the user | :black_circle: | :black_circle: | :black_circle: |
| Generate module report in Security events module with filter added by the Level 12 or above alerts KPI | :black_circle: | :black_circle: | :black_circle: |
| Generate module report in Security events module with filter added by the Authentication failure KPI | :black_circle: | :black_circle: | :black_circle: |
| AGenerate module report in Security events module with filter added by the Authentication success KPI | :black_circle: | :black_circle: | :black_circle: |
| Generate module report in Security events module with filter added by the Level 12 or above alerts KPI and agent.name in search bar input | :black_circle: | :black_circle: | :black_circle: |
| Generate module report in Integrity monitoring module without filters set by the user | :black_circle: | :black_circle: | :black_circle: |
| Generate agent inventory report and check the data for the vulnerabilities | :black_circle: | :black_circle: | :black_circle: |

**Details**
<details>
<summary>:black_circle: Generate module report in Security events module without filters set by the user</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Generate module report in Security events module with filter added by the Level 12 or above alerts KPI</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Generate module report in Security events module with filter added by the Authentication failure KPI</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: AGenerate module report in Security events module with filter added by the Authentication success KPI</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Generate module report in Security events module with filter added by the Level 12 or above alerts KPI and agent.name in search bar input</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Generate module report in Integrity monitoring module without filters set by the user</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Generate agent inventory report and check the data for the vulnerabilities</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

### Check List
- [ ] All tests pass
  - [ ] `yarn test:jest`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Comment by Desvelao** [comment]
**review**:

:green_circle: tests

Legend:
:black_circle:: none
:green_circle:: pass
:yellow_circle:: warning
:red_circle:: fail
:white_circle:: not applicable

## UI

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| Generate module report in Security events module without filters set by the user | :green_circle: | :black_circle: | :black_circle: |
| Generate module report in Security events module with filter added by the Level 12 or above alerts KPI | :green_circle: | :black_circle: | :black_circle: |
| Generate module report in Security events module with filter added by the Authentication failure KPI | :green_circle: | :black_circle: | :black_circle: |
| AGenerate module report in Security events module with filter added by the Authentication success KPI | :green_circle: | :black_circle: | :black_circle: |
| Generate module report in Security events module with filter added by the Level 12 or above alerts KPI and agent.name in search bar input | :green_circle: | :black_circle: | :black_circle: |
| Generate module report in Integrity monitoring module without filters set by the user | :green_circle: | :black_circle: | :black_circle: |

**Details**
<details>
<summary>:green_circle: Generate module report in Security events module without filters set by the user</summary>

Chrome - :green_circle:
image
image

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:green_circle: Generate module report in Security events module with filter added by the Level 12 or above alerts KPI</summary>

Chrome - :green_circle:
image
image

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:green_circle: Generate module report in Security events module with filter added by the Authentication failure KPI</summary>

Chrome - :green_circle:
image
image

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:green_circle: AGenerate module report in Security events module with filter added by the Authentication success KPI</summary>

Chrome - :green_circle:
image
image

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:green_circle: Generate module report in Security events module with filter added by the Level 12 or above alerts KPI and agent.name in search bar input</summary>

Chrome - :green_circle:
image
image

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:green_circle: Generate module report in Integrity monitoring module without filters set by the user</summary>

Chrome - :green_circle:
image
image

Firefox - :black_circle:

Safari - :black_circle:

</details>

File: server/lib/reporting/base-query.ts [hunk 1]
```diff
@@ -10,41 +10,26 @@
...
-    }
+    query: parsedserverSideQuery
   };
-
   //Add allowed agents filter
-  if(allowedAgentsFilter?.query?.bool){
+  if (allowedAgentsFilter?.query?.bool) {
     base.query.bool.minimum_should_match = allowedAgentsFilter.query.bool.minimum_should_match;
     base.query.bool.should = allowedAgentsFilter.query.bool.should;
   }
```
Comment (Desvelao) [inline server/lib/reporting/base-query.ts:35]: **thought**: if the parsedServerSideQuery contains the filters related to the allowed agents feature, we could be duplicating the query. I checked logging the query, see screenshot:

image

**Comment by yenienserrano** [comment]
### Test

Legend:
:black_circle:: none
:green_circle:: pass
:yellow_circle:: warning
:red_circle:: fail
:white_circle:: not applicable

## UI

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| Generate module report in Security events module without filters set by the user | :black_circle: | :black_circle: | :black_circle: |
| Generate module report in Security events module with filter added by the Level 12 or above alerts KPI | :black_circle: | :black_circle: | :black_circle: |
| Generate module report in Security events module with filter added by the Authentication failure KPI | :black_circle: | :black_circle: | :black_circle: |
| AGenerate module report in Security events module with filter added by the Authentication success KPI | :black_circle: | :black_circle: | :black_circle: |
| Generate module report in Security events module with filter added by the Level 12 or above alerts KPI and agent.name in search bar input | :black_circle: | :black_circle: | :black_circle: |
| Generate module report in Integrity monitoring module without filters set by the user | :black_circle: | :black_circle: | :black_circle: |
| Generate agent inventory report and check the data for the vulnerabilities | :black_circle: | :black_circle: | :black_circle: |

**Details**
<details>
<summary>:black_circle: Generate module report in Security events module without filters set by the user</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Generate module report in Security events module with filter added by the Level 12 or above alerts KPI</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Generate module report in Security events module with filter added by the Authentication failure KPI</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: AGenerate module report in Security events module with filter added by the Authentication success KPI</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Generate module report in Security events module with filter added by the Level 12 or above alerts KPI and agent.name in search bar input</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Generate module report in Integrity monitoring module without filters set by the user</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Generate agent inventory report and check the data for the vulnerabilities</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

**Comment by Desvelao** [comment]
### Test

Legend:
:black_circle:: none
:green_circle:: pass
:yellow_circle:: warning
:red_circle:: fail
:white_circle:: not applicable

## UI

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| Generate module report in Security events module without filters set by the user | 🟢 | :black_circle: | :black_circle: |
| Generate module report in Security events module with filter added by the Level 12 or above alerts KPI | 🟢 | :black_circle: | :black_circle: |
| Generate module report in Security events module with filter added by the Authentication failure KPI | 🟢 | :black_circle: | :black_circle: |
| AGenerate module report in Security events module with filter added by the Authentication success KPI | 🟢 | :black_circle: | :black_circle: |
| Generate module report in Security events module with filter added by the Level 12 or above alerts KPI and agent.name in search bar input | 🟢 | :black_circle: | :black_circle: |
| Generate module report in Integrity monitoring module without filters set by the user | 🟢 | :black_circle: | :black_circle: |
| Generate agent inventory report and check the data for the vulnerabilities | 🟢 | :black_circle: | :black_circle: |
| Generate module report in Policy monitoring module without filters set by the user | :black_circle: | :black_circle: | 
:black_circle: |
| Generate module report in PCI DSS module without filters set by the user | :black_circle: | :black_circle: | 
:black_circle: |
| Generate module report in GDPR module without filters set by the user | :black_circle: | :black_circle: | 
:black_circle: |
| Generate module report in TSC module without filters set by the user | :black_circle: | :black_circle: | 
:black_circle: |

**Details**
<details>
<summary> :green_circle: Generate module report in Security events module without filters set by the user</summary>

Chrome - :green_circle: 

image

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary> :green_circle: Generate module report in Security events module with filter added by the Level 12 or above alerts KPI</summary>

Chrome - :green_circle: 
image

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary> :green_circle:  Generate module report in Security events module with filter added by the Authentication failure KPI</summary>

Chrome - :green_circle:  
image

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary> :green_circle:  Generate module report in Security events module with filter added by the Authentication success KPI</summary>

Chrome - :green_circle: 
image

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary> :green_circle:  Generate module report in Security events module with filter added by the Level 12 or above alerts KPI and agent.name in search bar input</summary>

Chrome - :green_circle: 
image


Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>  :green_circle:  Generate module report in Integrity monitoring module without filters set by the user</summary>

Chrome - :green_circle:  
image

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary> :green_circle:  Generate agent inventory report and check the data for the vulnerabilities</summary>

Chrome - :green_circle: 
image

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary> :green_circle:  Generate module report in Policy monitoring module without filters set by the user</summary>

Chrome - :green_circle: 

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary> :green_circle:  Generate module report in PCI DSS module without filters set by the user</summary>

Chrome - :green_circle: 
image
image

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary> :green_circle:  Generate module report in GDPR module without filters set by the user</summary>

Chrome - :green_circle: 
image
image

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary> :green_circle:  Generate module report in TSC module without filters set by the user</summary>

Chrome - :green_circle: 
image
image

Firefox - :black_circle:

Safari - :black_circle:

</details>

**Review by Desvelao — APPROVED** [review]
LGTM!

**Comment by yenienserrano** [comment]
### Test

Legend:
:black_circle:: none
:green_circle:: pass
:yellow_circle:: warning
:red_circle:: fail
:white_circle:: not applicable

## UI

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| Generate module report in Security events module without filters set by the user | 🟢 | :black_circle: | :black_circle: |
| Generate module report in Security events module with filter added by the Level 12 or above alerts KPI | 🟢 | :black_circle: | :black_circle: |
| Generate module report in Security events module with filter added by the Authentication failure KPI | 🟢 | :black_circle: | :black_circle: |
| AGenerate module report in Security events module with filter added by the Authentication success KPI | 🟢 | :black_circle: | :black_circle: |
| Generate module report in Security events module with filter added by the Level 12 or above alerts KPI and agent.name in search bar input | 🟢 | :black_circle: | :black_circle: |
| Generate module report in Integrity monitoring module without filters set by the user | 🟢 | :black_circle: | :black_circle: |
| Generate agent inventory report and check the data for the vulnerabilities | 🟢 | :black_circle: | :black_circle: |
| Generate module report in Policy monitoring module without filters set by the user | :black_circle: | :black_circle: | 
:black_circle: |
| Generate module report in PCI DSS module without filters set by the user | :black_circle: | :black_circle: | 
:black_circle: |
| Generate module report in GDPR module without filters set by the user | :black_circle: | :black_circle: | 
:black_circle: |
| Generate module report in TSC module without filters set by the user | :black_circle: | :black_circle: | 
:black_circle: |

**Details**
<details>
<summary> 🟢 Generate module report in Security events module without filters set by the user</summary>

Chrome - 🟢 

image

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary> 🟢 Generate module report in Security events module with filter added by the Level 12 or above alerts KPI</summary>

Chrome - 🟢

image

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary> 🟢 Generate module report in Security events module with filter added by the Authentication failure KPI</summary>

Chrome - 🟢 

image

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary> 🟢 Generate module report in Security events module with filter added by the Authentication success KPI</summary>

Chrome - 🟢 

image

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary> 🟢 Generate module report in Security events module with filter added by the Level 12 or above alerts KPI and agent.name in search bar input</summary>

Chrome - 🟢 

image

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary> 🟢 Generate module report in Integrity monitoring module without filters set by the user</summary>

Chrome - 🟢 

image

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary> 🟢 Generate agent inventory report and check the data for the vulnerabilities</summary>

Chrome - 🟢 

image

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary> :green_circle:  Generate module report in Policy monitoring module without filters set by the user</summary>

Chrome - :green_circle: 
image
image

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary> :green_circle:  Generate module report in PCI DSS module without filters set by the user</summary>

Chrome - :green_circle: 
image
image

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary> :green_circle:  Generate module report in GDPR module without filters set by the user</summary>

Chrome - :green_circle: 
image
image

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary> :green_circle:  Generate module report in TSC module without filters set by the user</summary>

Chrome - :green_circle: 
image
image

Firefox - :black_circle:

Safari - :black_circle:

</details>

**Review by yenienserrano — APPROVED** [review]
LGTM

**Pull request merged**